File: src/main/java/at/hrechny/predictionsbot/database/entity/AuditEntity.java
package at.hrechny.predictionsbot.database.entity;

import at.hrechny.predictionsbot.database.model.ApiProvider;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Index;
import jakarta.persistence.Table;
import java.time.Instant;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@EqualsAndHashCode(callSuper = true)
@Table(name = "audit", indexes = {
    @Index(name = "idx_api_key_provider", columnList = "apiKey, apiProvider", unique = true)
})
public class AuditEntity extends GeneratedIdEntity {

  @Column
  private String apiKey;

  @Column
  @Enumerated(EnumType.STRING)
  private ApiProvider apiProvider;

  @Column
  private String requestUri;

  @Column(columnDefinition = "TIMESTAMP")
  private Instant requestDate;

  @Column
  private boolean success;

}

File: src/main/kotlin/at/hrechny/predictionsbot/database/entity/AuditEntity.kt
package at.hrechny.predictionsbot.database.entity

import io.micronaut.core.annotation.Introspected
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Index
import jakarta.persistence.Table
import java.time.Instant

@Introspected(accessKind = [Introspected.AccessKind.FIELD])
@Entity
@Table(
    name = "audit",
    indexes = [
        Index(name = "idx_audit_connector_code_request_date", columnList = "connector_code, request_date"),
    ],
)
class AuditEntity : GeneratedIdEntity() {
    @field:Column(name = "connector_code")
    var connectorCode: String? = null

    @field:Column(length = 2048)
    var requestUri: String? = null

    @field:Column(columnDefinition = "TIMESTAMP")
    var requestDate: Instant? = null

    @field:Column
    var success: Boolean = false

    @field:Column(columnDefinition = "TEXT")
    var failureReason: String? = null

    @field:Column(columnDefinition = "TEXT")
    var quotaSnapshot: String? = null

    fun isSuccess(): Boolean = success
}
